### PR TITLE
add jsonpath to handlers

### LINKED ISSUE #17: add jsonpath for handlers
to avoid passing (potentially) large state objects

### PR DESCRIPTION
**Pull request by ReinsBrain**
closes #17  - this needs some testing during review - i will be adding unit-tests for my next task

### CODE REVIEW

**Comment by jegj** [comment]
i will test it

**Review by jegj — CHANGES_REQUESTED** [review]

File: lib/store.js [hunk 1]
```diff
@@ -44,9 +44,21 @@ class Store extends EventEmitter {
 
 		const _dispatch = function ( action ) {
 			if ( _handlers[action.type] ) {
-				let trans = _handlers[action.type](action, Store.clone(_state));
-				if (trans.newState) {
-					_state = Store.clone(trans.newState);
+				let trans;
+				if ( _handlers[action.type].jsonpath ) {
```
[the remark this answers is not in the document]
Comment (pablolimo) [reply lib/store.js]: There we will have to get creative with the information provided in the action and put the listener on a top-level component let's say:

storage.dispatch({
type: 'UPDATE_ROW_ENTRY',
id: 'row1',
data: ...
});

Then on the list that holds the row entry do:

storage.on('UPDATE_ROW_ENTRY', (state, action) => {
const { id } = action;
// "UPDATE THE DESIRED ROW ENTRY"
});

[the remark this answers is not in the document]
Comment (jegj) [reply lib/store.js]: mmm i thought that ... but i think  maybe could be a better solution

**Comment by oreyes1991** [comment]
closing this because version 2 doesn't use JSONPath.

**Pull request closed without merging**